File: src/file.ts
import { invoke } from "@tauri-apps/api"

export async function getFileName(dir: string) {
    return await invoke('get_file_name', { dir: dir }).then(
        (v) => v
    ).catch((e) => { throw console.error(e) });
}

export async function getDirectoryName(dir: string) {
    return await invoke('get_directory_name', { dir: dir }).then(
        (v) => v
    ).catch((e) => { console.error(e) });
}

File: src-tauri/src/file.rs
use std::{path::Path, ffi::OsString};

#[tauri::command(rename_all = "snake_case")]
pub fn get_file_name(dir: String) -> String {  
    let path: &Path = Path::new(&dir);

    let path_name: String = String::from(
        path.file_name()
        .unwrap_or(&OsString::from(""))
        .to_str()
        .unwrap_or("")
    );

    return path_name;
}

#[tauri::command(rename_all = "snake_case")]
pub fn get_directory_name(dir: String) -> String {
    let path: &Path = Path::new(&dir);

    let path_parent: String = String::from(
        path.parent()
        .unwrap()
        .to_string_lossy()
    );

    return path_parent;
}